### ISSUE TITLE
Mod Feature: Remove content from "Flagged Items"

### ISSUE THREAD

**Issue by sizzlemctwizzle** [OpenUserJs/OpenUserJS.org#642]
The flag count would not be reset but instead` content.flagged = false;` (pardon). If the content gets flagged again, it would once again show up for review and possible removal/pardoning.

**Comment by Martii** [comment]
"Pardoning" seems like an interesting concept. :)

**Comment by Martii** [comment]
Just realized this is a dupe of #134 although it's better worded here! :) Closing the other issue.

**Comment by sizzlemctwizzle** [comment]
I created a nearly exact duplicate of another issue I created almost exactly a year later... I think I might be getting old...

**Comment by Martii** [comment]
Naw ... just getting wiser. :)

Questions:
1. Only allow pardoning if there aren't higher roles than yourself that have already flagged?
2. Can Admins+ pardon too or just a "capture the flag and toss it in file 13" like it is currently?
3. Track who pardons and perhaps how many times?
4. I assume any data stored would be in the `Flag` model or are we going to do this on `User`?
5. ...

**Comment by sizzlemctwizzle** [comment]
1. Admin+ can't flag so this wouldn't be an issue. However this would be an issue with who can bring something back after it had been removed. Although not that big of a deal since the group of users with roles above admin are fixed.
2. Admin+ can remove (sent to the graveyard) flags from content directly since they can actually inspect flags, unlike mods. Pardoning is a mod feature.
3. There should definitely be a record of pardoning.
4. I'd store this information on the content in the new `flags` object you added.

**Comment by Martii** [comment]
1) Still has the question... let's say I flag a user with a valid reason for long term watching... then it shows up in critical... and Moderator comes along and pardons it... I see an issue there.

**Comment by sizzlemctwizzle** [comment]
Okay, here is the problem. I never remember anyone with a role of admin or above being able to flag anything. In fact, I made a conscious decision not to allow this: see [my code](https://github.com/OpenUserJs/OpenUserJS.org/blob/0e2b026fdc6fd2ac0a300be17fd3189ceec797c7/libs/flag.js#L14) in a very early version of libs/flag.js (the module that controls the rules behind flagging.

I did some research and found out that apparently @Zren took it upon himself to silently [remove this logic](https://github.com/OpenUserJs/OpenUserJS.org/commit/df552c7a158c51a18998f7bfae71dac872e37fe8#diff-f968c831502e727188f4ce4cf56f755fL11), even though why it's there is clearly commented. It was a long time ago, and I guess I just never realized I **could** flag things.

I would like this reversed. Admins+ are not supposed to be part of the moderation system. They just take action when needed.

**Comment by Martii** [comment]
> Admins+ are not supposed to be part of the moderation system. They just take action when needed.

Well I can restore that part but I would still like to be able to long term track a user... but I can do that with my `Martii` account since that is moderator level... eventually I'll take action as Admin with `Marti` account but I usually like to give people a chance to make amends before I go killing their script/account.

Thanks for the insight and clarification.

**Comment by sizzlemctwizzle** [comment]
I think it's fine for you to have a mod account to flag things you want to
keep track of and give the author/user a chance to fix things. We don't
have any mods so you're kind of forced to fill the dual role of a mod/admin.

**Comment by Martii** [comment]
Well I still have that recommendation for a Moderator if he accepts but the Moderation tools **have to be complete** otherwise anyone won't bother... that's why I assigned myself to these issues... so hopefully we can find the right balance and understanding. I've been a sysop for decades on many other protocols so this isn't much extra work for me having dual roles... plus it helps me understand what the goal is/was. :)

**Comment by Martii** [comment]
Okay so followup question for 1) ...

5) Use case... if a script is flagged by anyone in the moderation system and Moderator 1 comes along and flags the same content... can a Moderator 2 pardon? e.g. equal role check?

**Comment by sizzlemctwizzle** [comment]
No. A moderator cannot pardon content flagged by another moderator. In a
situation like this, if the second moderator believes the first moderator
made a mistake by flagging, he/she should contact an admin to resolve the
situation.

**Comment by Martii** [comment]
@sizzlemctwizzle 

Re: 
> A button that lets moderators change a script in the flagged state to not flagged.

In order to pardon a moderator _must_ know the reason why a script is flagged. We can make the reporter invisible but without this info they can't pardon base off guessing why it was flagged.
